app/views/inthenewscommentaries/show.html.erb

```diff
@@ -1,51 +1,79 @@
 </div>
 
 <div class="well">
-  <%= sanitize @item.content %>
+  <%= sanitize @res["html_k"] %>
 </div>
 
-<% if @item.events.present? %>
-  <h3>Events</h3>
-  <ul>
-    <% @item.events.each do |rel| %>
-      <% id = db_to_es_id("event", rel.id) %>
-      <li><%= link_to rel.name, inthenews_events_item_path(id: id) %></li>
-    <% end %>
-  </ul>
+<% if @res["subjects"].present? %>
+  <h3>Events (<%= @res["subjects"].count %>)</h3>
+  <div class="item_data">
+    <ul>
+      <% @res["subjects"].each do |subject| %>
+        <%# markdown parsing %>
+        <% name = /\[(.*)\]/.match(subject)[1] if /\[(.*)\]/.match(subject) %>
+        <% id = /\]\((.*)\)/.match(subject)[1] if /\]\((.*)\)/.match(subject) %>
+        <li><%= link_to name, inthenews_events_item_path(id) %></li>
+      <% end %>
+    </ul>
+  </div>
 <% end %>
 
-<% if @item.news_items.present? %>
-  <h3>News Items</h3>
-  <ul>
-    <% @item.news_items.each do |rel| %>
-      <% id = db_to_es_id("news", rel.id) %>
-      <li><%= link_to rel.name, inthenews_news_items_item_path(id: id) %></li>
-    <% end %>
-  </ul>
+<% if @res["medium"].present? %>
+  <h3>News Items (<%= @res["medium"].count %>)</h3>
+  <div class="item_data">
+    <ul>
+      <% @res["medium"].each do |item| %>
+        <%# markdown parsing %>
+        <% name = /\[(.*)\]/.match(item)[1] if /\[(.*)\]/.match(item) %>
+        <% id = /\]\((.*)\)/.match(item)[1] if /\]\((.*)\)/.match(item) %>
+        <% if name %>
+          <% if id %>
+            <li><%= link_to name, inthenews_news_items_item_path(id) %></li>
+          <% else %>
+            <li><%= name %></li>
+          <% end %>
+        <% end %>
+      <% end %>
+    </ul>
+  </div><%# / .item-data %>
 <% end %>
 
-<% if @item.people.present? %>
-  <h3>People</h3>
-  <ul>
-    <% @item.people.each do |rel| %>
-      <% id = db_to_es_id("person", rel.id) %>
-      <li><%= link_to rel.name, inthenews_poets_item_path(id: id) %></li>
-    <% end %>
-  </ul>
+<% if @res["person"].present? %>
+  <h3>People (<%= @res["person"].count %>)</h3>
+  <div class="item_data">
+    <ul>
+      <% @res["person"].each do |person| %>
+        <% name = person["name"] %>
+        <% id = person["id"]  %>
+        <% if name %>
+          <% if id %>
+            <li><%= link_to name, inthenews_news_items_item_path(id) %></li>
+          <% else %>
+            <li><%= name %></li>
+          <% end %>
+        <% end %>
+      <% end %>
+    </ul>
+  </div><%# / .item-data %>
 <% end %>
 
-<% if @item.works.present? %>
-  <h3>Works</h3>
-  <ul>
-    <% @item.works.each do |rel| %>
-      <% id = db_to_es_id("work", rel.id) %>
-      <li><%= link_to rel.name, inthenews_works_item_path(id: id) %></li>
-    <% end %>
-  </ul>
+<% if @res["works"].present? %>
+  <h3>Related Works (<%= @res["works"].count %>)</h3>
+  <div class="item_data">
+    <ul>
+      <% @res["works"].each do |work| %>
+        <%# markdown parsing %>
+        <% name = /\[(.*)\]/.match(work)[1] %>
```

Comment:
I really hope in the future we can be more systematic abut this (call a library or a helper method. Not messing with it for this though.

Reply:
I agree. I think this would be best situated as a helper, as they are generally recommended for keeping logic out of `views/` templates and partials. But this could be made into an issue for later as I think it'd take a bit more time to get right for all the places this kind of code is used.

Reply:
made an issue

Reply:
Commented and retitled the issue